examples/platforms/gp712/uart-socket.c

```diff
@@ -0,0 +1,348 @@
+#define SOCKET_READ(socketId, buf, length)          recv(socketId, buf, length, 0)
+
+typedef struct
+{
+    uint32_t socketId;
+    bool isValid;
+    struct sockaddr addr;
+    pthread_t rfReadThread;
+} PlatSocket_t;
+
+PlatSocket_t   PlatSocketConnection;
+int PlatSocketPipeFd [2];
+int PlatServerSocketId;
+
+void PlatSocketRxNewConn(uint8_t id);
+void PlatSocketInit(void);
+void PlatSocketDeInit(void);
+int PlatSocketTxData(uint16_t length, uint8_t *pData, uint32_t socketId);
+
+#define PLAT_UART_MAX_CHAR 1024
+
+uint32_t PlatSocketId = 0;
+
+void PlatSocketSendInput(void* buffer)
+{
+    uint8_t len = 0;
+    uint8_t* buf = (uint8_t*) buffer;
+    len = strlen((char*)buf);
+    otPlatUartReceived((uint8_t*) buf, (uint16_t)len);
+    free(buf);
+    buf = 0;
+    len = 0;
+}
+
+void PlatSocketRx(uint16_t length, const char* buffer, uint32_t socketId)
+{
+    uint8_t* buf = 0;
+    PlatSocketId = socketId;
+    if (length > 0)
+    {
+        buf = malloc(length+2);
+        memcpy(buf, buffer, length);
+        buf[length]   = '\n';
+        buf[length+1] = 0;
+        qorvoAlarmScheduleEventArg(0, PlatSocketSendInput, (void*) buf);
+    }
+}
+
+void PlatSocketClose(uint32_t socketId)
+{
+    (void)socketId;
+}
+
+otError otPlatUartEnable(void)
+{
+    PlatSocketInit();
+    return OT_ERROR_NONE;
+}
+
+otError otPlatUartDisable(void)
+{
+    PlatSocketDeInit();
+    return OT_ERROR_NONE;
+}
+
+otError otPlatUartSend(const uint8_t *aBuf, uint16_t aBufLength)
+{
+    otError error = OT_ERROR_NONE;
+    char localbuf[PLAT_UART_MAX_CHAR];
+
+    memcpy(localbuf, aBuf, aBufLength);
+    localbuf[aBufLength] = 0;
+    printf("%s", localbuf);
+
+    if(PlatSocketId)
+    {
+        PlatSocketTxData(aBufLength, (uint8_t*) aBuf, PlatSocketId);
+    }
+    otPlatUartSendDone();
+    return error;
+}
+
+void platformUartInit(void)
+{
+}
+
+void platformUartProcess(void)
+{
+}
+
+int PlatSocketListenForClients()
+{
+    //Setup server side socket
+    int sockfd;
+    struct sockaddr_in serv_addr;
+    uint32_t flag = 1;
+    int32_t ret;
+
+    if ((sockfd = socket(AF_INET,SOCK_STREAM,0)) < 0)
+    {
+        perror("SetupListener: Error creating socket");
+        return -1;
+    }
+
+    // disable Nagle's algorithm to avoid long latency
+    ret = setsockopt(sockfd, IPPROTO_TCP, TCP_NODELAY, (char *)&flag, sizeof(flag));
+    // allow reuse of the same address
+    ret = setsockopt(sockfd, SOL_SOCKET,SO_REUSEADDR, (char *)&flag,sizeof(flag));
+    memset(&serv_addr, 0, sizeof(serv_addr));
+
+    serv_addr.sin_addr.s_addr = INADDR_ANY;
+    serv_addr.sin_family = AF_INET;
+    serv_addr.sin_port = htons(SOCKET_PORT);
+    if(bind(sockfd, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0)
+    {
+        perror("SetupListener: Error binding socket");
+        goto PlatSocketListenForClients_error;
```

Comment:
As per style guide, please dont use goto.  Instead, use the provided error handling macros